src/helix/client/client_ext.rs

```diff
@@ -105,30 +123,101 @@ impl<'client, C: crate::HttpClient + Sync + 'client> HelixClient<'client, C> {
     /// # let token = twitch_oauth2::AccessToken::new("validtoken".to_string());
     /// # let token = twitch_oauth2::UserToken::from_existing(&client, token, None, None).await?;
     /// use twitch_api::{helix, types};
+    /// use futures::TryStreamExt;
     ///
-    /// let ids: &[&types::UserIdRef] = &["1234".into(), "4321".into()];
     /// let chatters: Vec<helix::channels::ChannelInformation> = client
-    ///    .get_channels_from_ids(ids, &token).await?;
+    ///    .get_channels_from_ids(&["1234", "4321"][..].into(), &token).try_collect().await?;
     /// # Ok(()) }
     /// ```
-    pub async fn get_channels_from_ids<'b, T>(
+    pub fn get_channels_from_ids<T>(
         &'client self,
-        ids: impl AsRef<[&types::UserIdRef]> + Send,
-        token: &T,
-    ) -> Result<Vec<helix::channels::ChannelInformation>, ClientError<C>>
+        ids: &'client types::Collection<'client, types::UserId>,
+        token: &'client T,
+    ) -> impl futures::Stream<Item = Result<helix::channels::ChannelInformation, ClientError<C>>>
+           + Send
+           + Unpin
+           + 'client
     where
         T: TwitchToken + Send + Sync + ?Sized,
     {
-        let ids = ids.as_ref();
-        if ids.len() > 100 {
-            return Err(ClientRequestError::Custom("too many IDs, max 100".into()));
-        }
-        self.req_get(
-            helix::channels::GetChannelInformationRequest::broadcaster_ids(ids),
-            token,
-        )
-        .await
-        .map(|response| response.data)
+        futures::stream::iter(ids.chunks(100).collect::<Vec<_>>())
+            .map(move |c| {
+                let req = helix::channels::GetChannelInformationRequest::broadcaster_ids(c);
+                futures::stream::once(self.req_get(req, token)).boxed()
+            })
+            .flatten_unordered(None)
+            .map_ok(|resp| futures::stream::iter(resp.data.into_iter().map(Ok)))
+            .try_flatten_unordered(None)
+    }
+
+    /// Get multiple [Stream](helix::streams::Stream)s from user ids.
+    ///
+    /// # Examples
+    ///
+    /// ```rust, no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    /// # let client: helix::HelixClient<'static, twitch_api::client::DummyHttpClient> = helix::HelixClient::default();
+    /// # let token = twitch_oauth2::AccessToken::new("validtoken".to_string());
+    /// # let token = twitch_oauth2::UserToken::from_existing(&client, token, None, None).await?;
+    /// use twitch_api::{types, helix};
+    /// use futures::TryStreamExt;
+    ///
+    /// let live: Vec<helix::streams::Stream> = client.get_streams_from_ids(&["123456", "987654"][..].into(), &token).try_collect().await?;
```

Comment:
```suggestion
    /// let live: Vec<helix::streams::Stream> = client
    ///     .get_streams_from_ids(&["123456", "987654"][..].into(), &token)
    ///     .try_collect()
    ///     .await?;
```

Not sure why rustfmt doesn't wrap here. It would also be fine to just wrap to two lines, but this is now consistent with the other method.

Reply:
https://github.com/rust-lang/rustfmt/issues/5627